### PR TITLE
Actually set impliedNodeFormat in more cases

### PR DESCRIPTION
**Pull request by andrewbranch**
A bad merge conflict resolution in #58825 caused it not to take effect. This finishes the job.

### CODE REVIEW

**Comment by andrewbranch** [comment]
@typescript-bot test it

File: tests/baselines/reference/impliedNodeFormatEmit2(module=commonjs).js [hunk 1]
```diff
@@ -43,10 +43,7 @@ Object.defineProperty(exports, "__esModule", { value: true });
 exports._ = void 0;
 exports._ = 0;
 //// [b.mjs]
-"use strict";
-Object.defineProperty(exports, "__esModule", { value: true });
-exports._ = void 0;
-exports._ = 0;
+export var _ = 0;
```
Comment (jakebailey) [inline tests/baselines/reference/impliedNodeFormatEmit2(module=commonjs).js:46]: This seems weird given `module=commonjs`?

Comment (jakebailey) [reply tests/baselines/reference/impliedNodeFormatEmit2(module=commonjs).js:46]: Oh this is .mjs

**Review by jakebailey — APPROVED** [review]

**Comment by jakebailey** [comment]
It's a shame this missed the beta... but I guess we should fix this oops now since we said we were going to do it?

File: src/compiler/program.ts [hunk 2]
```diff
@@ -1392,16 +1392,13 @@ export function getImpliedNodeFormatForFileWorker(
...
-                fileExtensionIsOneOf(fileName, [Extension.Dcts, Extension.Cts, Extension.Cjs]) ? ModuleKind.CommonJS :
-                fileExtensionIsOneOf(fileName, [Extension.Dts, Extension.Ts, Extension.Tsx, Extension.Js, Extension.Jsx]) ? lookupFromPackageJson() :
-                undefined; // other extensions, like `json` or `tsbuildinfo`, are set as `undefined` here but they should never be fed through the transformer pipeline
-        default:
-            return undefined;
-    }
+    const moduleResolution = getEmitModuleResolutionKind(options);
+    const shouldLookupFromPackageJson = ModuleResolutionKind.Node16 <= moduleResolution && moduleResolution <= ModuleResolutionKind.NodeNext
+        || pathContainsNodeModules(fileName);
+    return fileExtensionIsOneOf(fileName, [Extension.Dmts, Extension.Mts, Extension.Mjs]) ? ModuleKind.ESNext :
```
Comment (jakebailey) [inline src/compiler/program.ts:1398]: Not for this PR but we copy and paste the arrays passed into `fileExtensionIsOneOf` all over the place; makes me wonder if we'd get a speed boost by lifting these arrays into common variables.

**Pull request merged**